x-pack/plugins/rollup/server/lib/search_strategies/rollup_search_strategy.js

```diff
@@ -0,0 +1,67 @@
+/*
+* Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+* or more contributor license agreements. Licensed under the Elastic License;
+* you may not use this file except in compliance with the Elastic License.
+*/
+import { indexBy, isString } from 'lodash';
+import { callWithRequestFactory } from '../call_with_request_factory';
+import mergeCapabilitiesWithFields from '../merge_capabilities_with_fields';
+import { getCapabilitiesForRollupIndices } from '../map_capabilities';
+
+const ROLLUP_INDEX_CAPABILITIES_METHOD = 'rollup.rollupIndexCapabilities';
+const batchRequestsSupport = false;
+
+const getRollupIndices = rollupData => Object.keys(rollupData);
+
+const isIndexPatternContainsWildcard = indexPattern => indexPattern.includes('*');
+const isIndexPatternValid = indexPattern => indexPattern &&
+  isString(indexPattern) && !isIndexPatternContainsWildcard(indexPattern);
+
+export default (AbstractSearchStrategy, RollupSearchRequest, RollupSearchCapabilities) =>
+  (class RollupSearchStrategy extends AbstractSearchStrategy {
+    name = 'rollup';
+
+    constructor(server) {
+      super(server, callWithRequestFactory, RollupSearchRequest);
+    }
+
+    getRollupData(req, indexPattern) {
+      const callWithRequest = this.getCallWithRequestInstance(req);
+
+      return callWithRequest(ROLLUP_INDEX_CAPABILITIES_METHOD, {
+        indexPattern,
+      }).catch(() => Promise.resolve({}));
```

Comment:
Are we silently catching any error on any rollup call? does we have any notification of a wrong call/es status etc somewhere else?

Reply:
It's ok, we just resolve our promise with {} if we cannot find rollup job by passed indexPattern. In this case we use default search strategy.